src/history/hs_rec.c

```diff
@@ -517,7 +517,7 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
 
             /* Detect any update without a timestamp. */
             if (prev_upd != NULL && prev_upd->start_ts < upd->start_ts) {
-                WT_ASSERT(session, prev_upd->start_ts == WT_TS_NONE);
+                WT_ASSERT_ALWAYS(session, prev_upd->start_ts == WT_TS_NONE, "out-of-order updated detected");
```

Comment:
out of order timestamp update detected?

Reply:
The third argument of `WT_ASSERT_ALWAYS` is `failure reason`. 
If `prev_upd->start_ts < upd->start_ts` is `true` and `prev_upd->start_ts == WT_TS_NONE` isn't true then we are looking at an OOO update. That is the reason the failure reason is out of `order timestamp update detected`.

Reply:
There is another place we need to add the check

```
    WT_ERR(ret);

    /*
     * Fail the eviction if we detect any timestamp ordering issue and the error flag is set. We
     * cannot modify the history store to fix the update's timestamps as it may make the history
     * store checkpoint inconsistent.
     */
    if (error_on_ts_ordering) {
        ret = EBUSY;
        WT_STAT_CONN_INCR(session, cache_eviction_fail_checkpoint_no_ts);
        goto err;
    }
```

Reply:
@kommiharibabu  @quchenhao  It doesn't seem straightforward to add an assertion here, do you think putting an assertion here is worth the effort, as we will fail before the function is even called?

Reply:
We should fix the name of this function __wt_hs_delete_key_from_ts, it doesn't take ts as input.

From the code, we can add an assert here like:

ts == 1 || ts == WT_TS_NONE

Reply:
Thanks, updated.